build-tasks/create-rule.js

```diff
@@ -0,0 +1,78 @@
+const common = require('./common');
+const fs = require('fs');
+
+const ruleName = getRuleName();
+
+const ruleFile = camelCase(ruleName) + 'Rule';
+const sourceFileName = 'src/' + ruleFile + '.ts';
+const testsFolder = 'tests/' + ruleName;
+const testFile = testsFolder + '/test.ts.lint';
+const lintFile = testsFolder + '/tslint.json';
+
+validateAguments();
+createImplementationFile();
+createTestFiles();
+addToConfig();
+
+console.log('Rule created');
+console.log('Rule source: ' + sourceFileName);
+console.log('Test file: ' + testFile);
+
+function getRuleName() {
+    const option = process.argv.find(str => str.startsWith('--rule-name'));
+
+    if (!option) {
+        return;
+    }
+
+    return option.split('=')[1];
+}
+
+function camelCase(input) {
+    return input.toLowerCase().replace(/-(.)/g, (match, group1) => group1.toUpperCase());
+}
+
+function validateAguments() {
+    const USAGE_EXAMPLE = '\nUsage example:\nnpm run create-rule -- --rule-name=no-something-or-other\n';
+
+    if (!ruleName) {
+        console.log('--rule-name parameter is required.' + USAGE_EXAMPLE);
+        process.exit(1);
+    }
+
+    if (!/^[a-z0-9]+(\-[a-z0-9]+)*$/.test(ruleName)) {
+        console.log('Rule name should consist of lowercase letters and numbers separated with "-" character.' + USAGE_EXAMPLE);
+        process.exit(1);
+    }
+}
+
+function createImplementationFile() {
+    const walkerName = ruleFile.charAt(0).toUpperCase() + ruleFile.substr(1) + 'Walker';
+
+    const ruleTemplate = require('./templates/rule.template');
+    const ruleSource = ruleTemplate({ruleName, walkerName});
+
+    common.writeFile(sourceFileName, ruleSource);
+}
+
+function createTestFiles() {
+    const testContent = '// Code that should be checked by rule';
+    const tslintContent = {
+        rules: {
+            [ruleName]: true
+        }
+    };
+
+    fs.mkdirSync(testsFolder);
```

Comment:
It should create folder with `0o777` permission according to Node.js [docs](http://nodejs.cn/doc/node/fs.html#fs_fs_mkdir_path_mode_callback).
Is it OK or what mode value should be used?

Reply:
Yeah this should be fine.